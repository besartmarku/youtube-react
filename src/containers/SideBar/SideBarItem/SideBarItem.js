import React from "react";
import { Icon, Menu } from "semantic-ui-react";
import "./SideBarItem.scss";

export function SideBarItem(props) {
  // React will ignore custom boolean attributes, therefore we pass a string
  // we use this attribute in our SCSS for styling
  const highlight = props.highlight ? "highlight-item" : null;
  return (
    <Menu.Item className={["sidebar-item", highlight].join(" ")}>
      <div className="sidebar-item-alignment-container">
        <span>
          <Icon size="large" name={props.icon} />{" "}
        </span>
        <span>{props.label}</span>
      </div>
    </Menu.Item>
  );
}
