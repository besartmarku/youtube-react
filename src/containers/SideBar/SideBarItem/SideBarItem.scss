.sidebar-item {
  background: #f5f5f5;
  span {
    i.icon {
      margin-right: 20px;
      color: #888888;
    }
  }

  &.highlight-item {
    span {
      font-weight: 600;
    }

    i.icon {
      color: #ff0002;
    }
  }

  .sidebar-item-alignment-container {
    display: flex;
    align-items: center;
  }
  .ui.menu.fixed.side-nav {
    background-color: #f5f5f5;
    margin-top: 64px;
    overflow-y: auto;
    padding-bottom: 64px;

    .sidebar-item:hover {
      background: #ebebeb;
    }
  }

  .side-nav.ui.vertical.menu {
    width: 17rem;
  }
}
