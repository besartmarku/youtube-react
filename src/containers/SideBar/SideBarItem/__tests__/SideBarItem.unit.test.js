import React from "react";
import { shallow } from "enzyme";
import { SideBarItem } from "../SideBarItem";

describe("SideBarItem", () => {
  test("renders empty SideBarItem", () => {
    const wrapper = shallow(<SideBarItem />);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders highlighted SideBarItem that navigates to /feed/trending", () => {
    const wrapper = shallow(
      <SideBarItem highlighted icon="fire" label="Trending" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("render non-highlighted SideBarItem that navigates to /feed/trending", () => {
    const wrapper = shallow(<SideBarItem icon="fire" label="Trending" />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Renders highlighted SideBarItem with no navigation", () => {
    const wrapper = shallow(
      <SideBarItem highlighted icon="fire" label="Trending" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
