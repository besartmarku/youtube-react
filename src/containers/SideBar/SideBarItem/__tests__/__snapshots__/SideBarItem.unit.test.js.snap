// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarItem Renders highlighted SideBarItem with no navigation 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
       
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;

exports[`SideBarItem render non-highlighted SideBarItem that navigates to /feed/trending 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
       
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;

exports[`SideBarItem renders empty SideBarItem 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        size="large"
      />
       
    </span>
    <span />
  </div>
</MenuItem>
`;

exports[`SideBarItem renders highlighted SideBarItem that navigates to /feed/trending 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
       
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;
